pvlib/solarposition.py

```diff
@@ -1210,3 +1210,84 @@ def hour_angle(times, longitude, equation_of_time):
     )).total_seconds() / 3600. for t in times])
     timezone = times.tz.utcoffset(times).total_seconds() / 3600.
     return 15. * (hours - 12. - timezone) + longitude + equation_of_time / 4.
+
+
+def _hours(times, hour_angle, longitude, equation_of_time):
+    """helper that converts hour angles in degrees into hours"""
+    tz_info = times.tz  # pytz timezone info
+    tz = tz_info.utcoffset(times).total_seconds() / 3600.
+    return (hour_angle - longitude - equation_of_time / 4.) / 15. + 12. + tz
+
+
+def _times(times, hours):
+    """helper converts hours from an array of floats to localized times"""
+    tz_info = times.tz
+    try:
+        hours = hours.values  # make a new reference
+    except AttributeError:
```

Comment:
I don't understand the point of this try/except. Is `hours` not always the same type? If not, suggest we change `_hours()` to do the right thing with something like 

```python
hours = # your eqn
# enforce Series
hours = pd.Series(hours, index=times)
# OR enforce array
hours = np.array(hours)
return hours
```

Then we can remove this try except.

Reply:
no apparently depending on the pandas version it changes, sometimes it's a numpy array of floats and sometimes it's a pandas series of `pandas.Float64Index`, I didn't bother to track down which versions are which, but it seems to be in the 0.teen versions, I think >0.20 yields series. I'm sure I could enforce the type somewhere else, as an output of some calculation, but I didn't bother to

Reply:
can you figure out where hours becomes a series? I'm not sure where it happens?